solutions/balkan-11-timeismoney.mdx

```diff
@@ -0,0 +1,186 @@
+---
+id: balkan-11-timeismoney
+source: Balkan OI 2011
+title: Timeismoney
+author: Chuyang Wang
+---
+
+[Official Analysis](http://www.boi2011.ro/resurse/tasks/timeismoney-sol.pdf)
+
+## Explanation
+
+For the 40% $t = c$ cases, we can solve the problem by constructing an MST with e.g. $t$ as the weights for the edges. Let $T = \sum t$ and $C = \sum c$. $V = T \cdot C = T^2$ will be minimized when $T$ is minimized. 
+
+To solve the problem completely, we will have to minimize the product of those two sums, $V = C \cdot T$. The solution space contains all possible MSTs with their respective $T$ and $C$ values. We can view each of those possible solutions as a 2D-point $(T, C)$. The inverse proportion function $C = \frac{V}{T}$ describes the hyperbola on which all the points have the same $V$ value. This hyperbola is convex, so it suffices to check all points on the lower convex hull of our solution space to find the minimal $V$ value. Nevertheless, we cannot really calculate the convex hull since it would require all $\binom{M}{N-1}$ points to be known. Instead, we use another approach below. 
```

Comment:
what point are you trying to make?

Reply:
so each possible MST has a C and T (and V=C*T) value, which can be plotted as a 2D point (T, C). There exists a curve (or hyperbola) C(T) = V / T s. t. every point on it (representing a MST) has the same V = C(T) * T = V / T * T =  V value. As a inverse proportion function, this curve (hyperbola) is convex, so the best solution must lie on the left most curve, i.e. somewhere on the (lower) convex hull.

Reply:
that's just...restating what you put there

Reply:
and the confusing part is <...> ?

Reply:
you know what, sure
i've no clue how convex hull works